src/process_utility.c

```diff
@@ -746,6 +746,156 @@ process_rename_index(Cache *hcache, Oid relid, RenameStmt *stmt)
 	}
 }
 
+/* Used as a tuple found function */
+static bool
+rename_hypertable_schema_name(TupleInfo *ti, void *data)
+{
+	char	  **schema_names = (char **) data;
+	HeapTuple	tuple = heap_copytuple(ti->tuple);
+	FormData_hypertable *ht = (FormData_hypertable *) GETSTRUCT(tuple);
+
+	/*
+	 * Because we are doing a heap scan with no scankey, we don't know which
+	 * schema name to change, if any
+	 */
+	if (namestrcmp(&ht->schema_name, schema_names[0]) == 0)
+		namestrcpy(&ht->schema_name, schema_names[1]);
+	if (namestrcmp(&ht->associated_schema_name, schema_names[0]) == 0)
+		namestrcpy(&ht->associated_schema_name, schema_names[1]);
+	if (namestrcmp(&ht->chunk_sizing_func_schema, schema_names[0]) == 0)
+		namestrcpy(&ht->chunk_sizing_func_schema, schema_names[1]);
+
+	catalog_update(ti->scanrel, tuple);
+	heap_freetuple(tuple);
+	/* Return true to keep going so we can change the name for all hypertables */
+	return true;
+}
+
+/* Used as a tuple found function */
+static bool
+rename_chunk_schema_name(TupleInfo *ti, void *data)
+{
+	HeapTuple	tuple = heap_copytuple(ti->tuple);
+	FormData_chunk *chunk = (FormData_chunk *) GETSTRUCT(tuple);
+
+	/* Rename schema name */
+	namestrcpy(&chunk->schema_name, (char *) data);
+	catalog_update(ti->scanrel, tuple);
```

Comment:
Hm, can you explain this behavior some more.. this callback *should* only fire if `chunk->schema_name == old_schema_name`. But it sounds like you're saying it might fire if `chunk->schema_name == new_schema_name`?

So suppose I have a single tuple (A="a", B="b"), and I'm renaming A from "a" to "c".  The first time the callback fires with tuple (A = "a", B="b"), and we rename the tuple to (A="c", B="b"). If I'm understanding you correctly, catalog_update has a weird behavior where the callback might fire _again_ with tuple (A="c", B="b")?

Reply:
The catalog_update() issue seems related to the fact that we call `CommandCounterIncrement()` on every update. The issue described by @cevian can probably be avoided by only calling `CommandCounterIncrement()` once at the end or when otherwise necessary (i.e., when the result of the update needs to be visible).

If necessary, we can expose alternative update functions or a boolean to optionally make changes visible.

Reply:
Hm that makes sense, but I'm still unclear what behavior we are seeing, since even if the update is "visible", the updated version doesn't match the scankey, so the callback shouldn't fire again?